packaging/greenboot/microshift-running-check.sh

```diff
@@ -103,6 +103,9 @@ function any_pods_running() {
 # Main
 #
 
+# export function for subshell access
+export -f microshift_service_active
```

Comment:
While this provides a solution for the current problem, the `functions.sh` code is generic, so this local solution is not generic enough. 

While, it is possible to write a generic code in BASH checking if a string is a function and it's exported, I'm not sure if we can add this code in `functions.sh` because we do not know if some part of the command is an unexported function.
```
if type "string_name" | grep -q 'function'; then
  echo "It is a function."
else
  echo "Not a function."
fi

if declare -F | grep -q 'function_name'; then
  echo "The function is exported."
else
  echo "The function is not exported."
fi
```

Reply:
Can we think about some other way to implement the timeout that does not complicate the function handling?

Reply:
Would something like this be better?
```
command_to_run &
pid=$!
sleep 5s && kill -HUP $pid &   # 5 seconds timeout
wait $pid
```

Reply:
> 

this wont work, because `wait `  only work if that pid is a child process of the current process,and timeout is invoked in a new shell.
```
function test(){ echo "calling function"; sleep 5; echo done; }
{ test & pid=$!; echo "waiting for ${pid}"; timeout 30 wait ${pid}; }

wait: pid 2865116 is not a child of this shell
```

but we can do something like  this:
```
function test(){ echo "calling function"; sleep 5; echo done; }
{ test & pid=$!; echo "waiting for ${pid}"; timeout 30 tail --pid=$pid -f /dev/null; }
```